web_programming/instagram_crawler.py

```diff
@@ -0,0 +1,176 @@
+            return info
+
+    def get_followers(self):
+        """
+        return number of followers
+        """
+
+        info = self.get_json()
+        followers = info['edge_followed_by']['count']
+        return followers
+
+    def get_followings(self):
+        """
+        return number of followings
+        """
+
+        info = self.get_json()
+        following = info['edge_follow']['count']
+        return following
+
+    def get_posts(self):
+        """
+        return number of posts
+        """
+
+        info = self.get_json()
+        posts = info['edge_owner_to_timeline_media']['count']
+        return posts
+
+    def get_biography(self):
+        """
+        return biography of user
+        """
+
+        info = self.get_json()
+        bio = info['biography']
+        return bio
+
+    def get_fullname(self):
+        """
+        return fullname of the user
+        """
+
+        info = self.get_json()
+        fullname = info['full_name']
+        return fullname
+
+    def get_username(self):
+        """
+        return the username of the user
+        """
+
+        info = self.get_json()
+        username = info['username']
+        return username
+
+    def get_profile_pic(self):
+        """
+        return the link of profile picture
+        """
+
+        info = self.get_json()
+        pic = info['profile_pic_url_hd']
+        return pic
+
+    def get_website(self):
+        """
+        return the users's website link
+        """
+
+        info = self.get_json()
+        external_url = info['external_url']
+        return external_url
+
+    def get_email(self):
+        """
+        return the email id of user if 
+        available
+        """
+
+        info = self.get_json()
+        return info['business_email']
+
+    def is_verified(self):
+        """
+        check the user is verified
+        """
+
+        info = self.get_json()
+        return info['is_verified']
+
+    def is_private(self):
+        """
+        check user is private
+        """
+
+        info = self.get_json()
+        return info['is_private']
+
+
+def html_1(soup):
+    """
+    parse the html type-1 of instagram
+    page
+    """
+    
+    scripts = soup.find_all('script')
+    main_scripts = scripts[4]
+    data = main_scripts.contents[0]
+    info_object = data[data.find('{"config"'):-1]
+    info = json.loads(info_object)
+    info = info['entry_data']['ProfilePage'][0]['graphql']['user']
+    return info
+
+
+def html_2(soup):
+    """
+    if html_1 fails, html_2 in action
+    parse the html type-2 of instagram
+    page
+    """
+    scripts = soup.find_all('script')
+    main_scripts = scripts[3]
+    data = main_scripts.contents[0]
+    info_object = data[data.find('{"config"'):-1]
+    info = json.loads(info_object)
+    info = info['entry_data']['ProfilePage'][0]['graphql']['user']
+    return info
+
+
+user = Instagram('github')
+print(user.is_verified())
+print(user.get_biography())
```

Comment:
Code that is at global scope will be run by our Travis CI / pytest process as discussed in CONTRIBUTING.md.

```suggestion
if __name__ == '__main__':
    user = Instagram('github')
    print(f"{user.is_verified() = })
    print(f"{user.get_biography() = })
```